### PR TITLE
Create a version of Meteor platform without Blaze

### PR DESCRIPTION
**Pull request by stubailo**
Also, implement a static HTML package that just lets you define static content for `<head>` and `<body>`.

Add both of these to Todos.

### CODE REVIEW

File: examples/react-todos/client/routes.jsx [hunk 1]
```diff
@@ -39,7 +39,8 @@ Meteor.startup(function () {
       showFirstList();
     }
 
-    React.render(<Handler handles={ handles } />, document.body);
+    React.render(<Handler handles={ handles } />,
+      document.getElementById("app-container"));
```
Comment (avital) [inline examples/react-todos/client/routes.jsx]: Just skimming, but why can't we render to `document.body` in this case?

File: examples/react-todos/.meteor/packages [hunk 2]
```diff
@@ -13,5 +14,6 @@ reactive-var
 meteorhacks:npm
 cosmos:browserify
 
+static-html
```
Comment (avital) [inline examples/react-todos/.meteor/packages]: Should this be part of `meteor-platform-no-ui`?

File: examples/react-todos/.meteor/versions [hunk 3]
```diff
@@ -60,6 +59,7 @@ sha@1.0.3
 spacebars@1.0.6
 spacebars-compiler@1.0.6
 srp@1.0.3
+static-html@0.0.1
 templating@1.1.1
```
Comment (avital) [inline examples/react-todos/.meteor/versions]: Hmm, why is templating still here?

**Comment by avital** [comment]
Just skimmed, but looks promising!

Are you duplicating code from the templating package? Can we instead somehow share code?

File: examples/react-todos/client/routes.jsx (hunk 1 above)
Comment (AdamBrodzinski) [inline examples/react-todos/client/routes.jsx]: @avital one of the subtle issues with rendering directly into `document.body` is that 3rd party scripts (like analytics, social buttons, etc...) can inject nodes into `document.body` and React doesn't seem to like it. This can cause hard to track down bugs.

File: examples/react-todos/.meteor/packages (hunk 2 above)
Comment (stubailo) [inline examples/react-todos/.meteor/packages]: It depends on whether we want this to be React-specific, or also work for Angular. I was thinking Angular could then finally use the `.html` file extension. @Urigo, what do you think? If there was a `meteor-platform` without the HTML source handler, would that make the Angular experience better?

File: examples/react-todos/.meteor/versions (hunk 3 above)
Comment (stubailo) [inline examples/react-todos/.meteor/versions]: It's definitely not running inside the app, because then `static-html` wouldn't work since you can't have two handlers for the same extension.

File: examples/react-todos/.meteor/packages (hunk 2 above)
Comment (Urigo) [inline examples/react-todos/.meteor/packages]: @stubailo  the only reason we use .ng.html is to get something like that, so I guess our HTML source handler will probably be exactly like `static-html`

Comment (Urigo) [inline examples/react-todos/.meteor/packages]: from looking at the code it doesn't seem like `static-html` is React specific, or I'm missing something?

Comment (Urigo) [inline examples/react-todos/.meteor/packages]: another thing would be to create a client only package that we could serve from a static website and not from Meteor, but I guess that is out of scope (although might be important for React devs as well)

Comment (stubailo) [inline examples/react-todos/.meteor/packages]: Oh, does the .ng.html thing not do anything special with the Angular templates? I thought maybe you would want to have that act on the .html files directly.

Comment (stubailo) [inline examples/react-todos/.meteor/packages]: If static-html will work for React and Angular both, then it should probably be in the meteor platform

Comment (Urigo) [inline examples/react-todos/.meteor/packages]: Yes I think it should be inside.
Right now we use `ng.html` to minify and also add the template to `template-cache` but it will be better that it will be optional.
Developers could use regular HTML files and if they really want they would use `ng.html` for template cache. Right now everything is in template cache.

By the way, what about minification and other plugins?

https://github.com/Urigo/angular-meteor/blob/master/plugin/handler.js

Comment (stubailo) [inline examples/react-todos/.meteor/packages]: Since this is static HTML only, it doesn't have anything to minify? Not sure how it is related.

Comment (Urigo) [inline examples/react-todos/.meteor/packages]: I mean like https://github.com/kangax/html-minifier
but I guess we Meteor doesn't do it anyway so never mind...

Comment (stubailo) [inline examples/react-todos/.meteor/packages]: Well, there is always room for improvement! We could minify the HTML if necessary.

Comment (Urigo) [inline examples/react-todos/.meteor/packages]: It might be a good idea, what do you think?

Comment (Urigo) [inline examples/react-todos/.meteor/packages]: It's definitely not necessary, nice to have at most

Comment (stubailo) [inline examples/react-todos/.meteor/packages]: Yeah, I think that would be a separate project.

**Comment by stubailo** [comment]
@avital after talking to @Slava, I have discovered that the `templating` package might be going through a lot of changes in the next Meteor release. Since I would like to ship the `static-html` package in the next week (possibly even on Monday) I think it would be best to do a hacky code-duplicating version in the mean time, and then refactor when the batch plugin changes land.

The only other alternative I see is delaying shipping this package until batch plugins land.

**Comment by stubailo** [comment]
Rebased on newest stuff. I'm still not sure if I should add `static-html` to the `meteor-platform-no-ui`... would people want to add their own source handlers for `html`?

**Comment by avital** [comment]
I vote for `meteor-platform-no-ui` to /not/ include `static-html`. Since otherwise people will similarly complain that they can't remove our default html parsing with their own custom one of they want to.

**Comment by stubailo** [comment]
After talking with @avital, I've decided that it would be best to ship these packages as part of Meteor 1.2 (and not before) in order to take advantage of the ongoing work on build system improvements.
